app/models/geolocation.rb

```diff
@@ -20,6 +20,11 @@ class Geolocation < ActiveRecord::Base
 
   belongs_to :geolocatable, polymorphic: true
 
+  validates :lat, format: { with: %r{\A-?([1-8]?[0-9]
+    \.{1}\d{1,6}\z|90\.{1}0{1,6}\z)}x }
+  validates :long, format: { with: %r{\A-?((([1]?[0-7][0-9]|[1-9]?[0-9])
+    \.{1}\d{1,6}\z)|[1]?[1-8][0]\.{1}0{1,6}\z)}x }
+
```

Comment:
```suggestion
  validates :lat, numericality: { greater_than_or_equal_to:  -90, less_than_or_equal_to:  90 }
  validates :long, numericality: { greater_than_or_equal_to: -180, less_than_or_equal_to: 180 }
```
Macht dasselbe, gibt genauere validation error messages ("Geolocations lat muss kleiner oder gleich 90 sein" statt "Geolocations lat ist nicht gültig") und wäre in meinen Augen lesbarer, was denkt ihr?

Reply:
Man könnte sich dann auch überlegen, die Einschränkung nach schweizer Koordinaten hier hinein zu verschieben. Aber ich finde es eigentlich gar nicht schlecht wenn die Geolocation weltweit benutzbar ist, und nur im Fall der Felder auf der Abteilung auf die Schweiz eingeschränkt wird.

Reply:
> Macht dasselbe, gibt genauere validation error messages ("Geolocations lat muss kleiner oder gleich 90 sein" statt "Geolocations lat ist nicht gültig") und wäre in meinen Augen lesbarer, was denkt ihr?

Die regex-Validierung gibt eine zusätzliche Einschränkung auf die Dezimalstellen, zum Beispiel 45.1234567 oder 47 wären nicht erlaubt (fraglich, ob das nötig ist). Die Fehlermeldung könnte man auch manuell anpassen? Bei der Lesbarkeit bin ich absolut einverstanden :-)

> Man könnte sich dann auch überlegen, die Einschränkung nach schweizer Koordinaten hier hinein zu verschieben. Aber ich finde es eigentlich gar nicht schlecht wenn die Geolocation weltweit benutzbar ist, und nur im Fall der Felder auf der Abteilung auf die Schweiz eingeschränkt wird.

Genau, die Idee war, nur die Abteilungs-Geolocations auf die Schweiz einzuschränken. Potentiell könnten ja weitere Felder dieselbe Klasse verwenden, für Koordinaten welche nicht zwingend in der Schweiz sind.

Reply:
Ah, stimmt, die Dezimalstellen. Ich hätte die jetzt ignoriert, aber wenn es für den Pfadi-Finder wirklich wichtig ist könnte man immer noch eine zweite Validierung mit einem einfacheren Regex einführen:
```ruby
validates :lat, numericality: { greater_than_or_equal_to:  -90, less_than_or_equal_to:  90 }, format: { with: /^\d+\.?\d{0,2}$/ }
```
Bedenkt einfach, dass die Validierung bisher nicht aktiv war, und der Pfadi-Finder daher sowieso keine Garantie dafür hat, was die Leute bereits für Werte in diese Felder eingetragen haben.

Reply:
Die Fehlermeldung könnte man manuell anpassen, aber numericality unterscheidet schon mal zwei Fälle, was du mit nur einem Regex nicht hinbekommst ("Geolocations lat ist keine Zahl" vs "Geolocations lat muss kleiner oder gleich 90 sein")

Reply:
Ja stimmt, diese Unterscheidung ist wohl informativer für den User. Ich werde mit @Michael-Schaer besprechen, was für den Pfadifinder wirklich nötig ist, und entsprechende Änderungen vornehmen.

Reply:
Die Lösung von Cosinus mit Regex ist ja wirklich besser lesbar :)
Die zusätzliche Überprüfung auf 1-6 Kommastellen fände ich gut!